From 0ccfd293ff8d0eff923d4dbece87c97dd8e7a073 Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Tue, 5 Nov 2019 14:51:00 +0200
Subject: [PATCH] PONRTSYS-6267: ppv4 rx-dma robustness error and packet drop
 if pmac header exists with modification session

Issue occur due to wrong STW field used by the recipe as P1 for the INSEREP
execution unit.

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-hal/modifier/modifier_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ppv4/pp-hal/modifier/modifier_core.c b/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
index 4f06e55811e8..65b5c82812b5 100644
--- a/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
+++ b/drivers/net/ppv4/pp-hal/modifier/modifier_core.c
@@ -764,7 +764,7 @@ s32 mod_hdr_rplc(struct mod_recipe *rcp, u32 bce_off, s32 end_trim_layer)
 	/* SI1 = 0 */
 	eu.si_mux1    = 0;
 	/* STW1 = L2 offset from STW MUX */
-	eu.stw_mux1   = EU_STW_SEL_L2_OFF;
+	eu.stw_mux1   = EU_STW_SEL_ZERO;
 	/* stw0 = end_trim_layer from STW MUX */
 	eu.stw_mux0   = end_trim_layer;
 	/* header starts in offset bce_off of bce SI region */
